### ISSUE TITLE
PyTorch implementation SimCLR v1

### ISSUE THREAD

**Issue by Spijkervet** [google-research/simclr#59]
Hi Ting!

It's been on GitHub for a while, but I have never shared the implementation with you. Since I read you are accepting implementations, I thought I'd share mine written in PyTorch.

It only includes scores for CIFAR/STL-10, and with my limited capacity to train the model, it's hard to directly compare if the implementation is a 100% correct.

https://github.com/Spijkervet/SimCLR/

All best!

Jane

**Comment by chentingpc** [comment]
Thanks Jane, I'm happy to link your repo! For CIFAR-10, the number could be further improved but hyper-parameters may be a reason. For reference, this repo (https://github.com/leftthomas/SimCLR) gets 92% in 100 epochs with ResNet-50 (and also in pytorch).

**Comment by Spijkervet** [comment]
Hi Ting! I just compared the two approaches. In my approach, I did not normalise the data augmentations after transformations (I'm unsure if this actually centers the data like it's supposed to or rather skews / correlates it back to the original set). I also pre-trained on 224x224 (instead of 32x32 in the other approach) for 100 epochs with ResNet50 (batch size 128 due to GPU constraints).

I have pre-training running with the same parameters as leftthomas + normalisation, should be done in a few hours. Will let you know!

**Comment by sdesrozis** [comment]
@Spijkervet Hi ! Your resnet seems a bit different than @leftthomas version : https://github.com/leftthomas/SimCLR/blob/6cea71a89ba00bbfcca91f5f8ffb04a7b203e95c/model.py#L7 In your new settings, did you include this ? Thanks !

**Comment by bhat-prashant** [comment]
@Spijkervet were you able to reproduce CIFAR10 results stated in the paper?

**Comment by bhat-prashant** [comment]
> Hi Ting! I just compared the two approaches. In my approach, I did not normalise the data augmentations after transformations (I'm unsure if this actually centers the data like it's supposed to or rather skews / correlates it back to the original set). I also pre-trained on 224x224 (instead of 32x32 in the other approach) for 100 epochs with ResNet50 (batch size 128 due to GPU constraints).
> 
> I have pre-training running with the same parameters as leftthomas + normalisation, should be done in a few hours. Will let you know!

Were you able to reproduce results? What changes did you make that resulted in the improvement? Thanks.

**Comment by bhat-prashant** [comment]
You are running Linear Evaluation until 500 epochs. If you stop the Linear Evaluation midway by saving the best performing model, you in fact achieve 4% improvement.
